pkg/cmd/deadman/operator_staleness_check.go

```diff
@@ -0,0 +1,177 @@
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "OperatorStalenessChecker"),
+	}
+
+	configInformers.Config().V1().ClusterOperators().Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				clusterOperator := obj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				clusterOperator := newObj.(*configv1.ClusterOperator)
+				c.queue.Add(clusterOperator.Name)
+			},
+		},
+		1*time.Minute,
+	)
+
+	return c
+}
+
+func isCheckingForStaleness(clusterOperator *configv1.ClusterOperator) bool {
+	for _, condition := range clusterOperator.Status.Conditions {
+		if strings.HasPrefix(condition.Message, "Checking for stale status") {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (c *OperatorStalenessChecker) syncHandler(ctx context.Context, key string) error {
+	clusterOperator, err := c.clusterOperatorLister.Get(key)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if !isCheckingForStaleness(clusterOperator) {
+		delete(c.operatorToDeadlineForResponse, clusterOperator.Name)
+		return nil
+	}
+
+	now := time.Now()
+	if _, ok := c.operatorToDeadlineForResponse[clusterOperator.Name]; !ok {
+		c.operatorToDeadlineForResponse[clusterOperator.Name] = now.Add(c.durationAllowedForOperatorResponse)
+	}
+	deadlineForReset := c.operatorToDeadlineForResponse[clusterOperator.Name]
+	if now.Before(deadlineForReset) {
+		return nil
+	}
+
+	// if we're past the deadline and some messsages still indicate we're checking for staleness, every condition should
+	// be marked as unknown and indicated as stale.
+	clusterOperatorToWriteAsStale := clusterOperator.DeepCopy()
+	for i, condition := range clusterOperator.Status.Conditions {
+		switch condition.Type {
+		case configv1.OperatorDegraded:
+			// if the operator is not setting status, it is Degraded.
+			clusterOperatorToWriteAsStale.Status.Conditions[i].Status = configv1.ConditionTrue
+		case configv1.OperatorUpgradeable:
+			if condition.Status == configv1.ConditionFalse {
+				// if the cluster isn't upgradeable, be sure not to suddenly allow improper upgrades
+			} else {
+				clusterOperatorToWriteAsStale.Status.Conditions[i].Status = configv1.ConditionUnknown
+			}
+
+		default:
+			// The other conditions are Unknown.
+			clusterOperatorToWriteAsStale.Status.Conditions[i].Status = configv1.ConditionUnknown
+		}
+
+		if condition.Status != clusterOperatorToWriteAsStale.Status.Conditions[i].Status {
+			clusterOperatorToWriteAsStale.Status.Conditions[i].LastTransitionTime = metav1.Time{Time: now}
+		}
+		clusterOperatorToWriteAsStale.Status.Conditions[i].Reason = "OperatorFailedStalenessCheck"
+		clusterOperatorToWriteAsStale.Status.Conditions[i].Message = "Operator is not setting status.  Last status was: " + condition.Message
```

Comment:
"Operator is not responding" ?
"Operator is not updating status for longer than N minutes" ?

Reply:
> Do we want to include the last reason here as well?

Sure, I can add that.